API-strategie-extensies/Summary v1.0.md

```diff
@@ -44,6 +44,18 @@ Design rules have unique and permanent numbers. In the event of design rules bei
 * <a href="#api-50">API-50</a>: Use CORS to control access
 * <a href="#api-52">API-52</a>: Use OAuth 2.0 for authorization with rights delegation
 * <a href="#api-58">API-58</a>: Do not put any sensitive information in URIs when communicating over shared networks
+* <a href="#api-59">API-59</a>: Use spinal-case for path segments
+* <a href="#api-60">API-60</a>: Normalize characters with diacritics to their base characters for path segments
+* <a href="#api-61">API-61</a>: Do not explicitly indicate that a resource is an API
+* <a href="#api-62">API-62</a>: Do not use file extensions in path segments
+* <a href="#api-63">API-63</a>: Do not use nonstandard abbreviations as resource names
+* <a href="#api-64">API-64</a>: Do not use compound words for nested objects
+* <a href="#api-65">API-65</a>: Use enumerations only for fixed sets of values that will not change
+* <a href="#api-66">API-66</a>: Use UPPER_SNAKE_CASE for enumeration values
```

Comment:
I don't necessarily agree with this convention; it is very JAVA/.NET centric and I find that we describe lots of data and only very rarely do we see enums use UPPER_SNAKE_CASE apart from JAVA/.NET API's

Reply:
Yes, but so are all other decisions. If we would have chosen lower_snake_case for instance, it would also have been specific to another framework. Another option is to leave it open, but then why are we standardising anyways...

Reply:
het doel is m.i. niet standaardiseren, dat is een middel. Het doel is bijvoorbeeld kosten besparen, fouten voorkomen, innovatie versnellen...

Het advies om codes te gebruiken bij enumeratiewaarden i.p.v. tekst kan bijdragen aan die doelen. En goed nadenken of iets wel een enumeratie moet zijn. Bijvoorbeeld om breaking changes te voorkomen.

Zelfde geldt voor het niet gebruiken. van diacrieten in enumeratiewaarden (geeft slecht leesbare gegenereerde code)

Ik denk dat een landelijke naamgevingsconventie voor enumeratiewaarden niet zoveel bijdraagt aan die doelen. Als je de OAS gebruikt om code te genereren, zal die code generator de enumeratiewaarden dan niet omzetten naar de standaard naamgevingsconventie voor dat framework?

Reply:
@mtrimpe @fsamwel dank voor jullie feedback. In overeenstemming met de werkgroep laat ik het zo staan als het nu staat. Wat betreft de comment van de code generator vraag ik me af of het wenselijk is om een code generator je API specificatie te laten bepalen. Wel moeten we mijns inziens toe naar een generiek commentaar bij de ADR welke aangeeft dat de ADR regels overruled kunnen worden in het geval een (inter)nationale standaard wordt gevolgd welke niet compatibel is op bepaalde punten met de ADR.